### PR TITLE
Smart contracts integration

### PR DESCRIPTION
**Pull request by gdalyy**
Implemented http calls : 
- POST /wallet
- POST /dao
- POST /space/addMembers
- POST /space/leave
- POST /coin/issue
- POST /coin/mint
- POST /coin/transfer

### CODE REVIEW

File: calls/Coin.php [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                    'name' => $coinName,
+                    'symbol' => Utils::getCoinSymbol($coinName),
+                    'decimals' => Utils::COIN_DECIMALS
+                ]
+            ]);
+
+            if ($response->getStatusCode() != HttpStatus::CREATED) {
+                throw new HttpException(
+                    $response->getStatusCode(),
+                    'Could not create ethereum coin for this space, will fix this ASAP !'
```
Comment (bumi) [inline calls/Coin.php:64]: can we do some exception logging that we get informed about that? (maybe with https://sentry.io/) 
but nothing we do in this PR, I just thought about it.

File: calls/Coin.php [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    /**
+     * @param $event
+     * @throws GuzzleException
+     * @throws HttpException
+     */
+    public static function mintCoin($event)
+    {
+        $transaction = $event->sender;
+
+        if ($transaction instanceof Transaction) {
```
Comment (bumi) [inline calls/Coin.php:79]: you could do: 
```php
if (!($transaction instanceof Transaction)) {
return;
}
... 
```
so not the whole function needs to be in the if block.

**Review by bumi — APPROVED** [review]
looks great! good job!

File: calls/Coin.php (hunk 1 above)
Comment (gdalyy) [reply calls/Coin.php:64]: Yes sure, good idea , i'll do that in coming PRs

File: calls/Coin.php (hunk 2 above)
Comment (gdalyy) [reply calls/Coin.php:79]: :+1:

**Pull request merged**